internal/environments/environments_test.go

```diff
@@ -3,68 +3,230 @@
+				{
+					Name: "terra-juyang-opera-fish",
+				},
+				{
+					Name: "terra-juyang-maggot-sawfly",
+				},
+			},
+			expectedError: nil,
+			expectedEnvironment: Environment{
+				Name: "terra-juyang-maggot-sawfly",
+			},
+		},
 	}
 
 	// Testing Code
 	for _, testCase := range testCases {
-		t.Run(testCase.name, func(t *testing.T) {
-			testApp := initTestApp(t)
-			defer testutils.Cleanup(t, testApp.db)
+		suite.Run(testCase.name, func() {
+			testutils.BeforeIntegrationSubTestCleanup(suite.T(), suite.testApp.db)
 
-			newEnvironment, err := testApp.Environments.CreateNew(testCase.request)
+			// create all non-final environments as setup
+			for _, request := range testCase.requests[:len(testCase.requests)-1] {
+				suite.testApp.Environments.CreateNew(request)
+			}
 
-			assert.Equal(t, testCase.expectedEnvironment.Name, newEnvironment.Name)
-			assert.Equal(t, testCase.expectedError, err)
+			// create and test the last environment
+			newEnvironment, err := suite.testApp.Environments.CreateNew(testCase.requests[len(testCase.requests)-1])
+			assert.Equal(suite.T(), testCase.expectedEnvironment.Name, newEnvironment.Name)
+			assert.Equal(suite.T(), testCase.expectedError, err)
 		})
 	}
 }
 
-//
-// Test Environment Setup
-//
+func (suite *EnvironmentTestSuite) TestIntegrationEnvironmentGetByName() {
+	suite.Run("GetByName gets an environment by name", func() {
+		testutils.BeforeIntegrationSubTestCleanup(suite.T(), suite.testApp.db)
 
-// only load the Controller we care about
-type TestApplication struct {
-	Environments *EnvironmentController
-	db           *gorm.DB
+		suite.testApp.Environments.CreateNew(suite.goodEnvironmentRequest)
+
+		foundEnvironment, err := suite.testApp.Environments.GetByName(suite.goodEnvironmentRequest.Name)
+
+		assert.Equal(suite.T(), foundEnvironment.Name, suite.goodEnvironmentRequest.Name)
+
+		assert.Equal(suite.T(), nil, err)
+	})
+
+	suite.Run("GetByName returns error if not found", func() {
+		testutils.BeforeIntegrationSubTestCleanup(suite.T(), suite.testApp.db)
+
+		suite.testApp.Environments.CreateNew(suite.goodEnvironmentRequest)
+
+		foundEnvironment, err := suite.testApp.Environments.GetByName("this-doesnt-exist")
+
+		assert.Equal(suite.T(), foundEnvironment.Name, "")
+		assert.Equal(suite.T(), errors.New("record not found"), err)
+	})
 }
 
-func initTestApp(t *testing.T) *TestApplication {
-	dbConn := testutils.ConnectAndMigrate(t)
-	app := &TestApplication{
-		Environments: NewController(dbConn),
-		db:           dbConn,
-	}
+func (suite *EnvironmentTestSuite) TestIntegrationEnvironmentListAll() {
+	suite.Run("ListAll returns nothing", func() {
+		testutils.BeforeIntegrationSubTestCleanup(suite.T(), suite.testApp.db)
 
-	return app
+		foundEnvironments, err := suite.testApp.Environments.ListAll()
+
+		assert.Equal(suite.T(), len(foundEnvironments), 0)
+		assert.Equal(suite.T(), nil, err)
+	})
+
+	suite.Run("ListAll returns one Environment", func() {
```

Comment:
This is probably a better way to handle listAll integration tests than the complicated seeding I was trying to do :)